cognite/client/data_classes/data_modeling/instances.py

```diff
@@ -1192,3 +1263,133 @@ def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None =
             if "externalId" in resource["unit"]
             else UnitSystemReference.load(resource["unit"]),
         )
+
+
+@dataclass
+class TypePropertyDefinition(CogniteObject):
+    type: PropertyType
+    nullable: bool = True
+    auto_increment: bool = False
+    immutable: bool = False
+    default_value: str | int | dict | None = None
+    name: str | None = None
+    description: str | None = None
+
+    def dump(self, camel_case: bool = True, return_flat_dict: bool = False) -> dict[str, Any]:
+        output: dict[str, Any] = {}
+        if return_flat_dict:
+            dumped = self._flatten_dict(self.type.dump(camel_case), ("type",))
+            output.update({key: value for key, value in dumped.items()})
+        else:
+            output["type"] = self.type.dump(camel_case)
+        output.update(
+            {
+                "nullable": self.nullable,
+                "autoIncrement": self.auto_increment,
+                "defaultValue": self.default_value,
+                "name": self.name,
+                "description": self.description,
+                "immutable": self.immutable,
+            }
+        )
+        return output
+
+    def _flatten_dict(self, d: dict[str, Any], parent_keys: tuple[str, ...]) -> dict[str, Any]:
+        items: list[tuple[str, Any]] = []
+        for key, value in d.items():
+            if isinstance(value, dict):
+                items.extend(self._flatten_dict(value, (*parent_keys, key)).items())
+            else:
+                items.append((".".join((*parent_keys, key)), value))
+        return dict(items)
```

Comment:
Is this just to simplify the structure as "convenience functionality" for the users, or is it needed because of some to/from API translation?

Should probably live in utils & have a unit test or two.